src/components/layout/AppMenu.tsx

```diff
@@ -0,0 +1,49 @@
+"use client";
+
+import React, { useState } from "react";
+import { usePathname } from "next/navigation";
+import Appbar from "@/components/layout/Appbar";
+import Navbar from "@/components/layout/Navbar";
+import UserMenu from "@/components/layout/UserMenu";
+import Providers from "@/providers/Providers";
+
+export default function AppMenu({
+  children,
+  appbarItems
+}: {
+  children: React.ReactNode;
+  appbarItems: React.ReactNode;
+}) {
+  const [isNavbarOpen, setIsNavbarOpen] = useState(false);
+
+  const pathname = usePathname();
+
+  const toggleNavbar = () => {
+    setIsNavbarOpen(!isNavbarOpen);
+  };
+
+  return (
+    <>
+      {pathname === "/login" ? (
+        <div className="grow flex flex-col">
+          {children}
+        </div>
+      ) : (
+        <Providers>

```

Comment:
Ty Providers bych asi hodil spíš do Layout, pokud to půjde. Jinak je to super změna.

Reply:
Problém je v tom, že Providers mají obalovat jen tu část mimo login, takže tam musíš udělat podmínku na základě usePath a jakmile to uděláš, tak je to client komponenta. Nevím, jestli by to šlo očůrat, že bychom do tu path poslali do layoutu nějakou proměnou, ale jestli to chápu správně, tak jakmile tam začnu dělat conditional rendering, tak to je client component, takže tím nic nevyřešíme.

Reply:
Providers můžou obalovat cokoliv, to je cajk, můžeš to dát klidně úplně nahoru. Providers jsou klient komponenta, ale předáváš jim jen children, takže všechno vevnitř může nadále být server komponenta.